### PR TITLE
Update kord to snapshot version

### PR DESCRIPTION
**Pull request by AMereBagatelle**
I needed some of these features myself, figured I'd go ahead and attempt an update.

I tested all of the changes I made on my own bot, and did my best to thoroughly test everything else.

Short list of changes:
- Updated Kord to snapshot builds
- Updated Kotlin to 1.6.10
- Changed all references of asString to toString on Snowflakes

It says it won't merge automatically, I'm not absolutely sure why.  The main reason I could see is the giant change in the README which was automatically generated by the tooling.

### CODE REVIEW

**Comment by AMereBagatelle** [comment]
Update: managed to merge the two branches together, merge conflicts are gone

File: buildSrc/src/main/kotlin/Constants.kt [hunk 1]
```diff
@@ -5,9 +5,9 @@ object Constants {
     const val releasesRepoUrl = "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
     const val snapshotsRepoUrl = "https://oss.sonatype.org/content/repositories/snapshots/"
 
-    const val kotlin = "1.6.0"
-    const val dokka = "1.6.0"
-    const val kord = "0.8.0-M8"
+    const val kotlin = "1.6.10"
+    const val dokka = "1.5.31"
```
Comment (ItsAzaria) [inline buildSrc/src/main/kotlin/Constants.kt:9]: Downgrade for dokka?

Comment (AMereBagatelle) [reply buildSrc/src/main/kotlin/Constants.kt]: Must've somehow missed that in the merge.  Has been fixed.

**Comment by AMereBagatelle** [comment]
Choosing to close PR after discussion on DiscordKt discord server.

**Pull request closed without merging**